tests/test_compr.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief Set of tests to validate the different compression algorithms
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include <gtest/gtest.h>
+
+#include "always.h"
+#include "asciistring.h"
+#include "bufffileclass.h"
+#include "compressionmanager.h"
+#include "refpack.h"
+#include "zlibcompr.h"
+
+#define WRITE_DATA 0
+
+// Some special refpack mode that doesn't seem to be used by the game. Test it anyways :)
+TEST(compression, refpack_encode_quick)
+{
+    auto filepath = Utf8String(TESTDATA_PATH) + "/compr/uncompr.txt";
+    BufferedFileClass src_file(filepath.Str());
+    ASSERT_TRUE(src_file.Open(FM_READ));
+    size_t src_size = src_file.Size();
+    uint8_t *src_data = new uint8_t[src_size];
+    src_file.Read(src_data, src_size);
+
+    uint8_t *dst_data = new uint8_t[src_size + 8];
+    uint32_t dst_size = RefPack_Compress(dst_data, src_data, src_size, true);
+    EXPECT_GT(dst_size, 0);
+    EXPECT_LT(dst_size, src_size);
+
+    delete[] src_data;
+    delete[] dst_data;
+}
+
+class CompressionTest : public ::testing::TestWithParam<CompressionType>
+{
+public:
+    void SetUp() override { m_type = GetParam(); }
+
+    struct PrintToStringParamName
+    {
+        template<class ParamType> std::string operator()(const testing::TestParamInfo<ParamType> &info) const
+        {
+            auto type = static_cast<CompressionType>(info.param);
+            return CompressionManager::Get_Compression_FourCC(type);
+        }
+    };
+
+protected:
+    CompressionType m_type;
+};
+
+TEST_P(CompressionTest, encode)
+{
+    auto filepath = Utf8String(TESTDATA_PATH) + "/compr/uncompr.txt";
```

Comment:
I am not sure whether or not it is good to use `Utf8String` and `BufferedFileClass` in these compression test functions. It means these tests can fail if there is a problem in `Utf8String` & Co. If you intend to write separate tests for `Utf8String` and `BufferedFileClass`, then perhaps it would make sense to use `std::string` and `std::fstream` here instead. That way compression test is pure. Or we run tests for strings and files before this, so that they would fail before this.

Reply:
Done